src/sources.rs

```diff
@@ -0,0 +1,25 @@
+//! Sources of data we want to sync from.
+
+use anyhow::Result;
+use async_trait::async_trait;
+
+use crate::{zitadel::SourceDiff, Config};
+
+pub mod csv;
+pub mod ldap;
+pub mod ukt;
+
+/// A source of data we want to sync from.
+#[async_trait]
+pub trait Source {
+	/// Create a new source from the given configuration.
+	fn new(config: &Config) -> Result<Self>
```

Comment:
Well, it would be either an associated type in the trait `Source` for the `new` method, or not having the `new` method in the Trait. Opting for the latter.